### PR TITLE
Basic word regex word search

### PR DESCRIPTION
**Pull request by Alce002**

### CODE REVIEW

File: src/Main.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import java.util.regex.Pattern;
+
+public class Main {
+    public static void main(String[] args) {
+        if(args.length != 2) {
+            System.err.println("Missing required parameters: <pattern> <file>");
+            return;
+        }
+
+        Pattern pattern = Pattern.compile(args[0]);
```
Comment (MiddlewQ) [inline src/Main.java:14]: Missing input validation for pattern

Is using regex pattern overkill?

Comment (MiddlewQ) [inline src/Main.java]: Missing input validation for args[0]

Is using regex pattern overkill?

File: src/Main.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            BufferedReader br = new BufferedReader(new FileReader(args[1]));
+            for(String line; (line = br.readLine()) != null;) {
+                if(pattern.matcher(line).find()) {
+                    System.out.println(line);
+                }
+            }
+        } catch (FileNotFoundException e)  {
+            System.err.println("Could not find file: " + args[1]);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```
Comment (MiddlewQ) [inline src/Main.java]: inconsistent errorhandling, should function throw error or use System.err? Why catch and then throw again?

File: src/Main.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    public static void main(String[] args) {
+        if(args.length != 2) {
+            System.err.println("Missing required parameters: <pattern> <file>");
+            return;
+        }
+
+        Pattern pattern = Pattern.compile(args[0]);
+
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(args[1]));
```
Comment (MiddlewQ) [inline src/Main.java]: Hard to read, break into descriptive steps

File: src/Main.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.regex.Pattern;
+
+public class Main {
+    public static void main(String[] args) {
```
Comment (MiddlewQ) [inline src/Main.java]: Missing functionality: require 'search' as first argument. 

Should main handle all code? Consider moving code to separate methods.

File: src/Main.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        if(args.length != 2) {
+            System.err.println("Missing required parameters: <pattern> <file>");
+            return;
```
Comment (MiddlewQ) [inline src/Main.java]: Use exit codes instead of returning normally

**Review by MiddlewQ — CHANGES_REQUESTED** [review]
small fixes

File: src/Main.java (hunk 3 above)
Comment (Alce002) [reply src/Main.java]: Likely better to not have an extra reference. All reads should go through BufferedReader.

**Review by MiddlewQ — APPROVED** [review]
LGTM 👍

**Pull request merged**